### ISSUE TITLE
cls.random_sample

### ISSUE THREAD

**Issue by bengolder** [joke2k/faker#67]
I wanted to add a method to BaseProvider that allows for sampling n unique elements.
There are situations in which I want to grab several random things, but I want those results to be unique. I just forked and added this to my own fork, but I wanted to run it by you before making a pull request.

```python
    # in faker/provides/__init__.py BaseProvider
    @classmethod
    def random_sample(cls, array=('a','b','c'), number=2):
        """ Returns $number unique elements from $array"""
        return random.sample(array, number)
```

**Comment by zzzuzik** [comment]
+1

**Comment by prokaktus** [comment]
What about random `text`s? It would be useful in factories https://github.com/rbarrois/factory_boy, where we need to generate `unique` field.

**Comment by fcurella** [comment]
I've started an implementation for this on #358. @prokaktus @jamescooke let me know if it's enough for your needs

**Comment by jamescooke** [comment]
`random_sample_unique` looks good to me :+1:

**Comment by prokaktus** [comment]
@fcurella looks nice for me too :+1: But what about unique texts? I can implement it with generating lot of samples and remove duplicates, but it would be nice to have some kind of generator, which always return unique text (address, name, etc.). What do you think about such feature?

I want to use it in my factory for models, which have `unique` constraints.

I can try to implement this myself in PR, if you think it can be useful.

**Comment by fcurella** [comment]
@prokaktus I can see how that could be useful eg. in generating usernames, but I would address that on a separate PR. Feel free to submit when you have a chance :)

**Comment by fcurella** [comment]
Closed via #358
